configs/recommended.js

```diff
@@ -40,9 +40,16 @@ module.exports = {
     'no-console': ['warn', { allow: ['warn', 'error'] }],
     'eol-last': 'error',
     'no-multi-spaces': 'error',
-    'import/no-unresolved': ['error', { ignore: ['^ui/.*', '^plugins/.*'] }],
+    'import/no-default-export': 'off',
+    'import/no-unresolved': ['error', { ignore: ['^ui/.*', '^plugins/.*', 'ng_mock'] }],
     'no-multiple-empty-lines': ['error', { max: 2, maxEOF: 0, maxBOF: 0 }],
     'quotes': ['error', 'single', { 'avoidEscape': true }],
+    "no-duplicate-imports": "off",
+    "@typescript-eslint/no-duplicate-imports": "error",
+    "no-use-before-define": "off",
+    "@typescript-eslint/no-use-before-define": ["error", { "functions": false, "classes": false }],
+    "no-extra-semi": "off",
+    "@typescript-eslint/no-extra-semi": "error",
```

Comment:
How come we have these, I'd very much prefer to fix these instead of allowing them. The extra semicolon looks like an easy `eslint --fix`, the others I'd at the very least signal them as warnings.

Reply:
These eslint rules are _replaced_ with the `@tyepscript-eslint` versions because they are enhanced to support both JS _and_ TS. So Javascript will still be checked for duplicate imports, extra semicolons, etc., but using the `@typescript-eslint` implementation of those rules.

Reply:
Ah! Ok, good.